iothub_client/samples/iothub_client_device_twin_and_methods_sample/iothub_client_device_twin_and_methods_sample.c

```diff
@@ -32,170 +42,550 @@
+    char* vanity_plate;         // reported property
+} STATE;
 
-typedef struct CAR_STATE_TAG
+typedef struct CAR_TAG
 {
-    int32_t softwareVersion;        // reported property
-    uint8_t reported_maxSpeed;      // reported property
-    char* vanityPlate;              // reported property
-} CarState;
+    bool change_oil_reminder;   // desired/reported property
+    char* last_oil_change_date; // reported property
+    MANUFACTURER manufacturer;  // reported property
+    STATE state;                // desired/reported property
+} CAR;
+
+static CAR device_car = { .change_oil_reminder = false,
+                          .last_oil_change_date = "May 4, 2016",
+                          .manufacturer = { .make = "Fabrikam",
+                                            .model = "sedan",
+                                            .year = 2014 },
+                          .state = { .allowed_max_speed = 100,
+                                     .software_version = 1.1,
+                                     .vanity_plate = "1T1" } };
+
+// Functions
+static void create_and_configure_device_client(void);
+static void connect_device_client_send_and_receive_messages(void);
+static void disconnect_device_client(void);
+
+static void update_properties(CAR* new_car);
+static void send_reported_property();
+static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context);
+static void get_twin_async_callback(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payload, size_t size, void* user_context);
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_OVER_WEBSOCKETS
+static void get_twin_desired_properties_async_callback(DEVICE_TWIN_UPDATE_STATE update_state, IOTHUB_TWIN_RESPONSE_HANDLE twin_response, const unsigned char* payload, size_t size, void* user_context);
+#endif
+static void patch_twin_desired_properties_callback(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payload, size_t size, void* user_context);
+static void patch_twin_reported_properties_callback(int status_code, void* user_context);
+static int receive_method_callback(const char* method_name, unsigned char const* payload, size_t size, unsigned char** response, size_t* response_size, void* user_context);
+static bool parse_twin_document(const unsigned char* payload, CAR* out_car, int64_t* out_parsed_desired_version);
+static bool parse_twin_document_desired(const unsigned char* payload, CAR* out_car);
+static bool parse_desired_patch(const unsigned char* payload, CAR* out_car, int64_t* out_parsed_desired_version);
+static char* malloc_and_build_json_reported_property(void);
+
+/*
+ * This sample utilizes the Azure IoT Hub to get the device twin document, send a reported
+ * property message, and receive desired property messages all in CBOR data format. It also shows
+ * how to set the content type system property for C2D and telemetry messaging. After 10 attempts to
```

Comment:
```suggestion
 * how to set the CBOR content type system property for C2D and telemetry messaging. After 10 attempts to
```

Reply:
fixed and update all comments to reflect if-not-version and not CBOR